src/components/Submit.tsx

```diff
@@ -0,0 +1,88 @@
+import { addDoc } from "firebase/firestore";
+import React, { useContext } from "react";
+import { ACTION } from "../context/Actions";
+import { GlobalContext } from "../context/GlobalState";
+import { colref, db } from "../firebase";
+import NavBar from "./NavBar";
+
+function Submit() {
```

Comment:
Not sure what does the name of the component signifies. Suggesting it to change to something more appropriate.